legal-api/src/legal_api/resources/v1/business/business_filings.py

```diff
@@ -405,19 +407,35 @@ def _put_basic_checks(identifier, filing_id, client_request) -> Tuple[dict, int]
                      f'Illegal to attempt to create a duplicate filing for {identifier}.'},
                     HTTPStatus.FORBIDDEN)
 
+        filing_type = json_input.get('filing', {}).get('header', {}).get('name')
+        if not filing_type:
+            return ({'message': 'filing/header/name is a required property'}, HTTPStatus.BAD_REQUEST)
+
+        if filing_type not in [
+            Filing.FILINGS['incorporationApplication']['name'],
+            Filing.FILINGS['registration']['name']
+        ] and business is None:
+            return ({'message': 'A valid business is required.'}, HTTPStatus.BAD_REQUEST)
+
         return None, None
 
     @staticmethod
-    def _check_authorization(identifier, filing_json: str) -> Tuple[dict, int]:
-        action = ['edit']
+    def _check_authorization(identifier, filing_json: dict, business: Business) -> Tuple[dict, int]:
         filing_type = filing_json['filing']['header'].get('name')
-        if filing_type == 'courtOrder':
-            action = ['court_order']
-        elif filing_type == 'registrarsNotation':
-            action = ['registrars_notation']
-        elif filing_type == 'registrarsOrder':
-            action = ['registrars_order']
-        if not authorized(identifier, jwt, action=action):
+        sub_filing_type = None
+        if filing_type == 'restoration':
+            sub_filing_type = filing_json['filing'].get('restoration', {}).get('type')
```

Comment:
I think we can use this function to get the filing sub-type https://github.com/bcgov/lear/blob/bb887cbdcf94a123717b133ca7c1bc24c20a685e/legal-api/src/legal_api/models/filing.py#L746-L754